src/main/java/Duke.java

```diff
@@ -244,7 +249,7 @@ private void deleteTask(int id) {
     private void saveTasks() throws DukeException {
         try {
             FileWriter fw = new FileWriter(SAVE_LOCATION);
-            for (Task task : this.tasks) {
+            for (Task task : tasks) {
                 if (task instanceof Deadline) {
                     fw.append(String.format("D | %d | %s | %s\n",
```

Comment:
Similarly, quite a long and complicated String! I think it might be better to break it up into temporary variables instead!!